src/utils.cc

```diff
@@ -140,7 +140,7 @@ std::string realPath(const std::string &path) {
 bool normalizeFolder(std::string &path) {
   for (auto &[root, real] : g_config->workspaceFolders)
     if (real.size() && llvm::StringRef(path).startswith(real)) {
-      // extra memory allocation may happen when string concatenates
+      path.reserve(root.size() + path.size() - real.size());
       path.clear();
```

Comment:
```suggestion
      path.clear();
      path.reserve(root.size() + path.size() - real.size());
```
on second look, it would be better to put the clear before the reserve, so that if reserve allocates it does not need to copy the old data in the new.